### PR TITLE
feat: Add example with an `emit_all` event

### PR DESCRIPTION
**Pull request by jojobyte**
This adds another example that demonstrates emitting an event from the plugin init and listening for that event in the frontend code and using it to change the background color of the `body`.

main.rs
```rust
#![cfg_attr(
  all(not(debug_assertions), target_os = "windows"),
  windows_subsystem = "windows"
)]

use tauri::{Manager};

#[derive(Clone, serde::Serialize)]
struct Payload {
  args: Vec<String>,
  cwd: String,
}

fn main() {
  tauri::Builder::default()
    .plugin(tauri_plugin_single_instance::init(|app, argv, cwd| {
      println!("{}, {argv:?}, {cwd}", app.package_info().name);
      app.emit_all("single-instance", Payload { args: argv, cwd }).unwrap();
    }))
    .run(tauri::generate_context!())
    .expect("error while running tauri application");
}
```

index.html
```html
<html>
  <body>
    <div>Plugin example</div>
    <pre id="args"></pre>
    <script type="module">
      var argsEl = document.getElementById('args')

      function cliProtocolVal({ args: { color: { value }}}) {
        argsEl.innerHTML = value
        document.body.style.backgroundColor = value
      }
      function cliProtocolErr (err) {
        argsEl.innerHTML = JSON.stringify(err)
      }

      window.__TAURI__.cli.getMatches()
        .then(cliProtocolVal)
        .catch(cliProtocolErr);

      await window.__TAURI__.event.listen(
        'single-instance',
        ({event, ...eventObj}) => {
          console.log(event, eventObj)
          let args = eventObj.payload?.args

          if (args?.length > 1) {
            let color = args[1]
            argsEl.innerHTML = color
            document.body.style.backgroundColor = color
          }
        }
      )
    </script>
  </body>
</html>
```

Which allows you to change the background by providing a color as an argument on the command line

```sh
npm install
npm run tauri dev

# in a separate terminal run this
./src-tauri/target/debug/emit-event-app cyan
```

### CODE REVIEW

File: examples/emit-event/package.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "tauri": "tauri"
+  },
+  "author": "",
+  "license": "MIT",
+  "dependencies": {
+    "@tauri-apps/api": "1.0.0"
+  },
```
Comment (amrbashir) [inline examples/emit-event/package.json]: This is not needed since you're using the global __TAURI__

**Review by amrbashir — CHANGES_REQUESTED** [review]
Thank you, this is a helpful example. Just one small nitpick and I will merge it.

File: examples/emit-event/package.json (hunk 1 above)
Comment (jojobyte) [reply examples/emit-event/package.json]: Removed and Done

**Review by amrbashir — APPROVED** [review]
Thanks

**Pull request merged**